tests/tracer/test_gitmetadata.py

```diff
@@ -0,0 +1,131 @@
+
+    envdir = os.path.join(example, "run_env_dir")
+    cwd = os.getcwd()
+    python_path = os.getenv("PYTHONPATH", None)
+    try:
+        subprocess.check_output("pip install --target=" + envdir + " " + pkgfile, shell=True)
+        os.chdir(envdir)
+        os.environ["PYTHONPATH"] = os.getenv("PYTHONPATH", "") + os.pathsep + envdir
+        yield
+    finally:
+        os.chdir(cwd)
+        if python_path is not None:
+            os.environ["PYTHONPATH"] = python_path
+
+
+@pytest.mark.usefixtures("preapare_test_env")
+class GitMetadataTestCase(TracerTestCase):
+    @run_in_subprocess(
+        env_overrides=dict(
+            DD_MAIN_PACKAGE="mypackage",
+        )
+    )
+    def test_gitmetadata_from_package(self):
+        tracer = ddtrace.Tracer()
+        tracer.configure(writer=DummyWriter())
+        with tracer.trace("span") as s:
+            pass
+
+        assert s.get_tag("_dd.git.commit.sha") == os.getenv("SHA_VALUE")
+        assert s.get_tag("_dd.git.repository_url") == "https://github.com/companydotcom/repo"
+
+    @run_in_subprocess(
+        env_overrides=dict(
+            DD_TAGS="git.commit.sha:12345,git.repository_url:github.com/user/tag_repo",
+            DD_MAIN_PACKAGE_="gitmetadatapoc",
+        )
+    )
+    def test_gitmetadata_from_DD_TAGS(self):
+        tracer = ddtrace.Tracer()
+        tracer.configure(writer=DummyWriter())
+        with tracer.trace("span") as s:
+            pass
+
+        print("test_gitmetadata_from_DD_TAGS:", s.get_tags())
+        # must be from DD_TAGS
+        assert s.get_tag("_dd.git.commit.sha") == "12345"
+        assert s.get_tag("_dd.git.repository_url") == "github.com/user/tag_repo"
+        # must be not present in old tags
+        assert s.get_tag("dd.git.repository_url") is None
+        assert s.get_tag("dd.git.commit.sha") is None
+
+    @run_in_subprocess(
+        env_overrides=dict(
+            DD_TAGS="git.commit.sha:12345,git.repository_url:github.com/user/tag_repo",
+            DD_GIT_COMMIT_SHA="123456",
+            DD_GIT_REPOSITORY_URL="github.com/user/env_repo",
+            DD_MAIN_PACKAGE="gitmetadatapoc",
+        )
+    )
+    def test_gitmetadata_from_ENV(self):
+        tracer = ddtrace.Tracer()
+        tracer.configure(writer=DummyWriter())
+        with tracer.trace("span") as s:
+            pass
+
+        # must be from env variables
+        assert s.get_tag("_dd.git.commit.sha") == "123456"
+        assert s.get_tag("_dd.git.repository_url") == "github.com/user/env_repo"
+        # must be not present in old tags
+        assert s.get_tag("dd.git.repository_url") is None
+        assert s.get_tag("dd.git.commit.sha") is None
+
+    @run_in_subprocess(
+        env_overrides=dict(
+            DD_TAGS="git.commit.sha:12345,git.repository_url:github.com/user/tag_repo",
+            DD_GIT_COMMIT_SHA="123456",
+            DD_GIT_REPOSITORY_URL="github.com/user/env_repo",
+            DD_MAIN_PACKAGE="gitmetadatapoc",
```

Comment:
is this package name meant to be `mypackage`? (same for the other tests in this file).

if it is meant to be a package that does not exist, can we use a more obvious name like `DD_MAIN_PACKAGE=non_existent_pkg` ("ish")

Reply:
Yes! Good catch! (it was from the previous tests, we renamed it after refactoring)